packages/cli/src/ui/contexts/ScrollProvider.tsx

```diff
@@ -158,14 +253,68 @@ export const ScrollProvider: React.FC<{ children: React.ReactNode }> = ({
     }
   };
 
+  const handleMove = (mouseEvent: MouseEvent) => {
+    const state = dragStateRef.current;
+    if (!state.active || !state.id) return;
+
+    const entry = scrollablesRef.current.get(state.id);
+    if (!entry || !entry.ref.current) {
+      state.active = false;
+      return;
+    }
+
+    const boundingBox = getBoundingBox(entry.ref.current);
+    if (!boundingBox) return;
+
+    const { y } = boundingBox;
+    const { scrollTop, scrollHeight, innerHeight } = entry.getScrollState();
+
+    const thumbHeight = Math.max(
+      1,
+      Math.floor((innerHeight / scrollHeight) * innerHeight),
+    );
+    const maxScrollTop = scrollHeight - innerHeight;
+    const maxThumbY = innerHeight - thumbHeight;
+
+    if (maxThumbY <= 0) return;
+
+    const relativeMouseY = mouseEvent.row - y;
+    // Calculate the target thumb position based on the mouse position and the offset.
+    // We clamp it to the valid range [0, maxThumbY].
+    const targetThumbY = Math.max(
+      0,
+      Math.min(maxThumbY, relativeMouseY - state.offset),
+    );
+
+    const targetScrollTop = Math.round(
+      (targetThumbY / maxThumbY) * maxScrollTop,
+    );
+
+    entry.scrollBy(targetScrollTop - scrollTop);
```

Comment:
Since ScrollableList and the ScrollableEntry interface were updated to support scrollTo, it is recommended to use entry.scrollTo(targetScrollTop) if available, mirroring the logic implemented in handleLeftPress.